src/ssmfe/ssmfe.f90

```diff
@@ -2170,9 +2170,15 @@ subroutine ssmfe_inverse_srci_double_complex &
 
       k = ilaenv(1, 'ZHETRD', 'U', nep, -1, -1, -1)
       k = max(2*nep - 1, (k + 2)*nep) + 3*nep - 1    
+      allocate ( dwork(max(3*nep - 2, 1), 1), stat = inform%stat )
```

Comment:
Good catch! Can I just check: why `max(3*nep - 2, 1)` for the size?

Reply:
That is the required dimension of that `RWORK` argument according to the [LAPACK documentation](https://netlib.org/lapack/explore-html/df/d9a/group__complex16_h_eeigen_gaf7b790b3b89de432a423c9006c1cc1ac.html). Perhaps that is worth a comment? On the other hand, the calculation for `k` seems equally opaque.

Reply:
Ah I see, fine as is then I think.